src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -544,11 +544,19 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				return false;
 			}
 
+			// only keep track of input if prompt hasn't already been shown
+			if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT, StorageScope.GLOBAL, true) &&
+				// just a normal key with no modifier
+				event.key.match(/^[a-z0-9]$/i) && !event.ctrlKey && !event.shiftKey && !event.altKey) {
```

Comment:
I'm not sure whether this regex is safe for non-english languages...

Reply:
my guess is no. not sure how to address this